imports/api/Testing/testingconstants.js

```diff
@@ -590,11 +590,16 @@ function createBumped() {
       let newVars = JSON.parse(JSON.stringify(DETECTORS[place[0]]['variables']));
       newVars.push('var ' + relationship + ';');
 
+      let newRules = JSON.parse(JSON.stringify(DETECTORS[place[0]]['rules']));
```

Comment:
I didn't understand what this was doing when we first spoke about it, but do now after seeing it code. I think it would be good to add a few comments explaining why you have to modify the detector rules like this.